pilot/pkg/model/virtualservice.go

```diff
@@ -97,12 +97,23 @@ func vsHostMatches(vsHost string, importedHost host.Name, vs config.Config) bool
 }
 
 func resolveVirtualServiceShortnames(rule *networking.VirtualService, meta config.Meta) {
+	// for internal generated Gateway API virtual services, we don't need to resolve shortnames for hosts
+	skipTopLevelResolution := UseGatewaySemantics(config.Config{
```

Comment:
cannot we  skip all the below logic when it is gateway converted VS?

Reply:
Thanks, just checked, and we can skip all the below logic.